vaadin-spring/src/main/java/com/vaadin/flow/spring/security/RequestUtil.java

```diff
@@ -55,10 +55,32 @@ public boolean isFrameworkInternalRequest(HttpServletRequest request) {
         return HandlerHelper.isFrameworkInternalRequest(vaadinMapping, request);
     }
 
+    /**
+     * Checks whether the request targets an endpoint.
+     *
+     * @param request the servlet request
+     * @return {@code true} if the request is targeting an enpoint, {@code false}
+     *         otherwise
+     */
     public boolean isEndpointRequest(HttpServletRequest request) {
         if (endpointUtil != null) {
             return ((EndpointUtil) endpointUtil).isEndpointRequest(request);
         }
         return false;
     }
+
+    /**
+     * Checks whether the request targets an endpoint that is public, i.e. marked
+     * as @{@link com.vaadin.flow.server.connect.auth.AnonymousAllowed}.
+     *
+     * @param request the servlet request
+     * @return {@code true} if the request is targeting an anonymous enpoint,
+     *         {@code false} otherwise
+     */
+    public boolean isAnonymousEndpoint(HttpServletRequest request) {
+        if (endpointUtil != null) {
+            return ((EndpointUtil) endpointUtil).isAnonymousEndpoint(request);
```

Comment:
Maybe it would be cleaner to not repeat this slightly ugly cast, and instead have a private method returning nullable `EndpointUtil` or alternatively `Optional<EnpointUtil>`?

Reply:
Can't use `EndpointUtil` in methods or the project won't work if you exclude Fusion

Reply:
Right, I keep missing that.